libraries/thermocouple/Thermocouple.h

```diff
@@ -97,71 +100,95 @@ namespace Thermocouple {
     }
   }
 
-  // namespace Analog { //TMP36
-  //   // calibration = 25 C at 750mV
-  //
-  //   int DTRDY_PIN = 28;
-  //   int RESET_PIN = 29;
-  //
-  //   // first 7 bits for Arduino Wire
-  //   uint8_t ADC2_ADDR = 0b1001000;
-  //
-  //   float tempOverVoltageScale = 1 / 0.01;
-  //   float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
-  //
-  //   long rawRead;
-  //   float voltageRead;
-  //   float tempRead;
-  //
-  //   ADS1219 ads(DTRDY_PIN, ADC2_ADDR, NULL);
-  //
-  //   void init() {
-  //     // ads.begin();
-  //
-  //     ads.setConversionMode(CONTINUOUS);
-  //
-  //     ads.setVoltageReference(REF_EXTERNAL);
-  //
-  //     ads.setGain(GAIN_ONE);
-  //
-  //     ads.setDataRate(90);
-  //   }
-  // 
-  //   void readTemperatureData(float *data) {
-  //     rawRead = ads.readData(2); // thermocouple on AIN1 of ADC2
-  //     // Serial.println(rawRead);
-  //     // Serial.println(ads.readData(2));
-  //     // Serial.println(ads.readData(3));
-  //     voltageRead = (float) rawRead * (5.0 / pow(2,23));
-  //     Serial.println(voltageRead);
-  //     tempRead = ((voltageRead - voltageOffset) * tempOverVoltageScale) + tempOffset;
-  //     data[0] = tempRead;
-  //     data[1] = -1;
-  //   }
-  // }
+  namespace Analog { //TMP36
+
+    // calibration = 25 C at 750mV
+
+    ADS1219 ** _adcs;
+
+    vector<int> * _analogInNumMap;
+
+    int _numSensors;
+
+    float tempOverVoltageScale = 1 / 0.01;
+    float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
+
+    long rawRead;
+    float voltageRead;
+    float tempRead;
+
+    void init (int numSensors, vector<int> * analogInMap, ADS1219 ** adcs) {
+      _numSensors = numSensors;
+      _analogInNumMap = analogInMap; // since this variable is coming from the config, I don't think I need to copy it, since it's not going to change/ be deallocated
+      _adcs = _adcs;
+    }
+
+    // float *data is only of size 7 rn, ensure that we only expect <= 7 readings.
+    void readTemperatureData(float *data) {
+      int index = 0;
+      for (int i = 0; i < _numSensors; i++) {
+        int ainCounter = 0;
+        for (std::vector<int>::iterator it = _analogInNumMap[i].begin();
+             it != _analogInNumMap[i].end(); ++it ) {
+          if(*it != 0) {
```

Comment:
generalized data reads in the library. 

no reference to specific locations / functions of thermocouple here. That should all be maintained w/in config

Reply:
I guess one concern I have is about the latency that this will add to each sensor reading, since we'll have to iterate through all ADC channels every time. Especially if we add more ADCs, this would add unnecessary delay. Since upon initialization we'll know how many of each sensor we have, at the beginning we could iterate through the sparse ADC mapping once and save the ADC # & ch # for each sensor, then when reading data just use those, and save doing the unnecessary iteration on every read

Reply:
Or you could create the 2 arrays with the ADC # & ch # for each sensor in the config, and just pass those into the library, so that when you need to read in the library you can just iterate over the # of sensors & Ch ___ of ADC ___, and only have to do that the same # of times as sensors you have.